src/Azure.Functions.Cli/Actions/LocalActions/InitAction.cs

```diff
@@ -349,22 +351,33 @@ private static string LanguageSelectionIfRelevant(WorkerRuntime workerRuntime)
 
         private void ValidateTargetFramework()
         {
-            if (ResolvedWorkerRuntime == Helpers.WorkerRuntime.dotnetIsolated)
+            if (string.IsNullOrEmpty(TargetFramework))
             {
-                if (string.IsNullOrEmpty(TargetFramework))
+                if (ResolvedWorkerRuntime == Helpers.WorkerRuntime.dotnetIsolated)
                 {
-                    // Default to .NET 8 if the target framework is not specified
-                    // NOTE: we must have TargetFramework be non-empty for a dotnet-isolated project, even if it is not specified by the user, due to the structure of the new templates
                     TargetFramework = DefaultTargetFramework;
                 }
-                if (!TargetFrameworkHelper.GetSupportedTargetFrameworks().Contains(TargetFramework, StringComparer.InvariantCultureIgnoreCase))
+                else if (ResolvedWorkerRuntime == Helpers.WorkerRuntime.dotnet)
                 {
-                    throw new CliArgumentsException($"Unable to parse target framework {TargetFramework}. Valid options are \"net8.0\", \"net7.0\", \"net6.0\", and \"net48\"");
+                    TargetFramework = DefaultInProcTargetFramework;
+                }
+                else
+                {
+                    return;
                 }
             }
-            else if (!string.IsNullOrEmpty(TargetFramework))
+
+            var supportedFrameworks = ResolvedWorkerRuntime == Helpers.WorkerRuntime.dotnetIsolated
+                ? TargetFrameworkHelper.GetSupportedTargetFrameworks()
+                : TargetFrameworkHelper.GetSupportedInProcTargetFrameworks();
+
+            if (!supportedFrameworks.Contains(TargetFramework, StringComparer.InvariantCultureIgnoreCase))
+            {
+                throw new CliArgumentsException($"Unable to parse target framework {TargetFramework}. Valid options are {string.Join(", ", supportedFrameworks)}");
+            }
+            else if (ResolvedWorkerRuntime != Helpers.WorkerRuntime.dotnetIsolated && ResolvedWorkerRuntime != Helpers.WorkerRuntime.dotnet)
             {
-                throw new CliArgumentsException("The --target-framework option is supported only when --worker-runtime is set to dotnet-isolated");
+                throw new CliArgumentsException("The --target-framework option is supported only when --worker-runtime is set to dotnet-isolated or dotnet");
```

Comment:
I believe the same thing needs to be done on [Line 130](https://github.com/Azure/azure-functions-core-tools/blob/8b7fc502df9759926a0afc35864c13c3c101f63f/src/Azure.Functions.Cli/Actions/LocalActions/InitAction.cs#L130)

It would also be nice to align the target framework options list. To that end, can we reorder to mirror what's in that help text today? I think that may mean just a change in the list defined in the helper.

Reply:
Updated. Also added a new test for that.